spec/support/test_env.rb

```diff
@@ -48,19 +39,27 @@ def disable_mailer
     NotificationService.any_instance.stub(mailer: double.as_null_object)
   end
 
-  def enable_mailer
-    NotificationService.any_instance.unstub(:mailer)
+  # Clean /tmp/tests
+  def clean_test_path
+    tmp_test_path = Rails.root.join('tmp', 'tests', '**')
+
+    Dir[tmp_test_path].each do |entry|
+      # Keep gitlab-shell and gitlab-test
+      FileUtils.rm_rf(entry) unless File.basename(entry) =~ /\Agitlab-(shell|test)\z/
+    end
   end
 
   def setup_gitlab_shell
-    `rake gitlab:shell:install`
+    unless File.directory?(Rails.root.join(*%w(tmp tests gitlab-shell)))
```

Comment:
`master`:

```
gitlab master % time DISABLE_SPRING=true rspec spec/factories_spec.rb
Cloning into bare repository '/Users/tsigo/Code/rails/gitlab-development-kit/gitlab/tmp/tests/gitlab-test_bare'...
done.
................................................

Finished in 18.78 seconds
48 examples, 0 failures

Randomized with seed 56784

DISABLE_SPRING=true rspec spec/factories_spec.rb  22.02s user 6.94s system 98% cpu 29.524 total
```

`rs-testenv`:
gitlab rs-testenv % time DISABLE_SPRING=true rspec spec/factories_spec.rb
................................................

Finished in 7.54 seconds
48 examples, 0 failures

Randomized with seed 54243

DISABLE_SPRING=true rspec spec/factories_spec.rb  14.29s user 3.77s system 97% cpu 18.459 total
```

Nothing major, but not minor either if you're running specs outside of guard as I do.